### ISSUE TITLE
[HOLD for payment 2024-08-27] [$250] Clicking on new message doesn't navigate to the last message for 2nd comment link

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#45004]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.5-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue found when executing PR:** https://github.com/Expensify/App/pull/41962
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Start a chat with account B as account A
3. Send 200 messages (Or use this expensifail account [applausetester+nlinbx@applause.expensifail.com](mailto:applausetester+nlinbx@applause.expensifail.com)
)
4. Copy the link of the 70th comment and send it
5. Copy the link of the 150th comment and send it
6. Go to troubleshoot and clear cache and restart
7. Click on the first link sent on step 4
8. Click on the "New message" badge that appeared at the top of the chat
9. Confirm the page scrolls to the bottom of the comment history
10. Click on the second link sent on step 5.
11. Click on the "New message" badge once again

## Expected Result:
The report page scrolls down to the bottom of the comment history

## Actual Result:
The report page doesn't scroll down to the bottom of the comment history after clicking on the second link, and this time clicking on the first link and then clicking on the new message badge no longer scrolls to the bottom of the comment history as well.

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/9c1133b8-010f-4501-9b3d-949018d67465

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0188f7ddb71ba94170</li>
        <li>Upwork Job ID: 1810560415299355486</li>
        <li>Last Price Increase: 2024-07-23</li>
<li>Automatic offers: </li>
<ul>
<li>tsa321 | Contributor | 103330551</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @RachCHopkins</details>

**Comment by RachCHopkins** [comment]
Replicated - To be fair my "new message" button didn't take me to the recent message at any point.

**Comment by s77rt** [comment]
@nyomanjyotisa Your RCA is not correct. Why is `hasNewestReportAction` false if we have the new message button?

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by s77rt** [comment]
@tsa321 

> when the user presses the "new message" button (route change), the comparison (isLinkingToMessage && !isLinkedMessagePageReady) will always be false because isLinkingToMessage is false, preventing the remount of reportActionsView and causing the report to not display the last message.

What is exactly that we are looking for in the remount logic? i.e. why do we need a remount? The latest report action is already displayed.


https://github.com/user-attachments/assets/913b00cf-8b8d-41bc-b93a-8766aec57fdb


**Related:** In [`scrollToBottomAndMarkReportAsRead` ](https://github.com/Expensify/App/blob/98d8a5ae4bb52539669f79e007246417eb8c8f1e/src/pages/home/report/ReportActionsList.tsx#L415-L424) why do we need to navigate to the same report?

**Related:** I noticed that calling `reportScrollManager.scrollToBottom()` does not scroll to the bottom on the first call, on the second call it does even though there is no difference in the report actions (didn't get new report actions after the first call)


https://github.com/user-attachments/assets/b35de7c4-4db7-4903-a988-551c18165653

**Comment by s77rt** [comment]
@tsa321 

> The latest messages (i.e., messages 0-10) won’t be displayed in the report actions list. This makes hasNewestReportActions false, and clicking on the new message button will navigate to the report without the reportActionID parameter. This will remount the reportActionsView (because the shouldShowSkeleton of ReportScreen will be displayed briefly), and the list will displays from the newest message and 50 messages above it.


It's still not clear what went wrong. The same logic happens if we have report actions in onyx, yet the new message button works.


https://github.com/user-attachments/assets/fd7f345f-ac7d-4891-a609-79f41420ada8

**Comment by s77rt** [comment]
@tsa321 The root cause is still not identified. I got it that unmouting `ReportActionsView` and mounting it again makes this work but why? (there is no difference in the report actions). Perhaps due to a `useEffect` that runs on mount only? (and not on re-render)


https://github.com/user-attachments/assets/12bf02af-2119-4017-aa3c-5c8f9bc64c7a

**Comment by s77rt** [comment]
@tsa321 I see you mentioned [`listID`](https://github.com/Expensify/App/blob/08bb00b2dc515bb1da2e70f9b9eae9d522c3be50/src/pages/home/report/ReportActionsView.tsx#L148-L159) in your second alternative proposal. It is the culprit. As for the solution can we go with something like that:
```js
const prevReportActionID = usePrevious(reportActionID);
...
if (!reportActionID && !prevReportActionID) {
    // Keep the old list ID since we're not in the Comment Linking flow
    return listOldID;
}
```

**Comment by s77rt** [comment]
@tsa321 Do you still experience the pagination bug with the above solution?

**Comment by s77rt** [comment]
@tsa321 Let's go with that solution then

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/45004#issuecomment-2254297005)

**Comment by s77rt** [comment]
This is due payment

**Comment by RachCHopkins** [comment]
@s77rt I think you need to remove the reviewing label to deem it ready for payment.

**Comment by RachCHopkins** [comment]
Or maybe Melvin was supposed to. All the automation seems to have failed here.

**Comment by RachCHopkins** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@s77rt] Determine if we should create a regression test for this bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@RachCHopkins] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by RachCHopkins** [comment]
- @s77rt requires payment via NewDot manual request(Reviewer)
- @tsa321 requires payment [automatic offer](https://www.upwork.com/nx/wm/offer/103330551) (Contributor)

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/39040
- The offending PR has been commented on: https://github.com/Expensify/App/pull/39040#discussion_r1732369093
- A discussion in #expensify-bugs has been started: Not needed (unique bug)
- Determine if we should create a regression test for this bug: No, this feels low priority, this test should be enough for now https://expensify.testrail.io/index.php?/cases/view/2661784

**Comment by RachCHopkins** [comment]
Payment Summary:

Contributor: @tsa321  paid $250 via Upwork
Contributor+: @s77rt  ready to be paid $250 via newdot manual request

Upwork job [here](https://www.upwork.com/nx/wm/offer/103330551)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed.

**Comment by JmillsExpensify** [comment]
$250 approved for @s77rt
